packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -204,12 +204,26 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         // // Check gas right before the call to get total gas consumed by OVM transaction.
         // uint256 gasProvided = gasleft();
 
-        // Run the transaction, make sure to meter the gas usage.
-        (, bytes memory returndata) = ovmCALL(
-            _transaction.gasLimit - gasMeterConfig.minTransactionGasLimit,
-            _transaction.entrypoint,
-            _transaction.data
-        );
+        bytes memory returndata;
+        if (_isUpgrading()) {
+            (bool success, bytes memory ret) = ovmCALL(
+                _transaction.gasLimit - gasMeterConfig.minTransactionGasLimit,
+                0x420000000000000000000000000000000000000D,
```

Comment:
It's also a little bit odd that `_transaction.entrypoint` is totally ignored in this path. 
I think this opens up a race condition based grieffing attack, where normal transactions get played after a call to `approveTransactionBundle`, which would result in those transactions failing in the `L2ChugSplashDeployer`. 
That said, I don't really see a way to avoid that.

Reply:
> I think this opens up a race condition based grieffing attack

I think the best way to avoid this right now is to turn on a sequencer firewall for a few minutes before the upgrade and figure out how to best communicate upgrades to front-ends. I don't think there's a great way around this. We really *want* transactions to fail when the upgrade starts so we don't risk any weird mid-upgrade behavior.

Reply:
Agree that this code path could use some additional comments though.

Reply:
Added comments.